dot/sync/chain_sync.go

```diff
@@ -0,0 +1,792 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/common/optional"
+	"github.com/ChainSafe/gossamer/lib/common/variadic"
+)
+
+const (
+	// MAX_WORKERS is the maximum number of parallel sync workers
+	// TODO: determine ideal value
+	MAX_WORKERS = 4
+)
+
+var _ ChainSync = &chainSync{}
+
+//nolint
+type (
+	BlockRequestMessage  = network.BlockRequestMessage
+	BlockResponseMessage = network.BlockResponseMessage
```

Comment:
I just like having the aliases so we don't need to import `network` into every file

Reply:
It makes it really cumbersome to peek the contents of the type when it's subtyped like this.  I usually embed the struct when I need to add attributes and receiver functions, but otherwise I don't mind using other public types.

Reply:
ok I'll remove these then